packages/http/src/utils/__tests__/fakeIncomingMessage.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { FakeIncomingMessage } from '../fakeIncomingMessage';
+import * as typeIs from 'type-is';
```

Comment:
@brendarearden , I'd be happy to add more tests here if you know of other ways we need to "fake" `IncomingMessage` behavior, but I think `typeIs` and `hasBody` were the main ones.  What others did I miss?